components/builder-api-client/src/lib.rs

```diff
@@ -139,10 +142,45 @@ impl Client {
         let mut encoded = String::new();
         res.read_to_string(&mut encoded).map_err(Error::IO)?;
         debug!("Body: {:?}", encoded);
-        let rd: ReverseDependencies = serde_json::from_str(&encoded).map_err(Error::JSON)?;
+        let rd: ReverseDependencies = serde_json::from_str(&encoded).map_err(Error::Json)?;
         Ok(rd.rdeps.to_vec())
     }
 
+    /// Promote a job group to a channel
+    ///
+    /// # Failures
+    ///
+    /// * Remote API Server is not available
+    pub fn job_group_promote(
+        &self,
+        group_id: u64,
+        channel: &str,
+        token: &str,
+    ) -> Result<Vec<String>> {
+        debug!("Promoting job group {} to channel {}", group_id, channel);
+        let url = format!("jobs/group/{}/promote/{}", group_id, channel);
+        let res = self.add_authz(self.0.post(&url), token).send().map_err(
+            Error::HyperError,
+        )?;
+
+        if res.status != StatusCode::Ok {
+            return Err(err_from_response(res));
+        }
+
+        match decoded_response::<UnsuccessfulJobGroupPromote>(res)
```

Comment:
Can `decoded_response` be tweaked to handle empty responses? i.e., can we push _that_ behavior further down so every caller of `decoded_response` doesn't have to handle it themselves?

I take your point about not exposing weird internal quirks via the external API, though. I'm fine with adding some explanatory comments to the code as it is. It still feels not-quite-right to me, but we can revise our approach as needed in the future.